taichi/ui/common/gui_base.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <string>
+#include "taichi/ui/utils/utils.h"
+
+TI_UI_NAMESPACE_BEGIN
+
+class GuiBase {
```

Comment:
Dear imgui offers a lot of other components. I've only included a few of the most basic ones here. I'd like to keep things concise for now, and in a future PR, we could add all the fancy stuff we want.